.github/workflows/pin-requirements.yml

```diff
@@ -0,0 +1,48 @@
+name: Pin Requirements
+
+on:
+  workflow_dispatch:
+  workflow_call:
+
+permissions:
+  pull-requests: write
+  contents: write
```

Comment:
Let's move these permissions into the individual job that needs them (there's only one job in this case, so this is just for general best practice).

Reply:
(Also, a short comment explaining why each permission is present would be good!)